src/attr/_funcs.py

```diff
@@ -40,24 +44,36 @@ def asdict(
 
     ..  versionadded:: 16.0.0 *dict_factory*
     ..  versionadded:: 16.1.0 *retain_collection_types*
+    ..  versionadded:: 20.3.0 *value_serializer*
     """
     attrs = fields(inst.__class__)
     rv = dict_factory()
     for a in attrs:
         v = getattr(inst, a.name)
+        if value_serializer is not None:
+            v = value_serializer(inst, a, v)
         if filter is not None and not filter(a, v):
```

Comment:
Wouldn't it be better, if we ran the filters first to avoid unnecessary serialization? The order should be documented in any case.

Reply:
Good point will change and document this.